app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,21 @@
+class CommentsController < ApplicationController
+  def new
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = Comment.new(comment_params)
+    if @comment.save
+      redirect_to @comment.post
+    else
+      render :new, status: :unprocessable_entity
+    end
```

Comment:
- [ ] It will be great to ensure that the comment is created on behalf of the currently logged-in user.  Although your implementation will still create a comment, You have already defined the `current_user` in the application controller. So you should use that user instead of fetching the `author_id ` from the params. Kindly check the sample below ⬇️ 
```ruby
def create
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]

    if @comment.save
      redirect_to post_path(@comment.post)
    else
      @article = Post.find(params[:post_id])
      redirect_to post_path(@post)
    end
  end
```
```ruby
def comment_params
    params.require(:comment).permit(:text)
  end
```

Reply:
Done. Thanks for your help